server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -742,6 +745,30 @@ public synchronized void verifyIndexMetadata(IndexMetadata metadata, IndexMetada
         }
     }
 
+    @Override
+    public IndexShard createShard(
+            final ShardRouting shardRouting,
+            final PeerRecoveryTargetService recoveryTargetService,
+            final PeerRecoveryTargetService.RecoveryListener recoveryListener,
+            final RepositoriesService repositoriesService,
+            final Consumer<IndexShard.ShardFailure> onShardFailure,
+            final Consumer<ShardId> globalCheckpointSyncer,
+            final RetentionLeaseSyncer retentionLeaseSyncer,
+            final DiscoveryNode targetNode,
+            final DiscoveryNode sourceNode) throws IOException {
+        IndexService indexService = indexService(shardRouting.index());
+        RecoveryState recoveryState = indexService.createRecoveryState(shardRouting, targetNode, sourceNode);
+
+        return createShard(shardRouting,
+            recoveryState,
+            recoveryTargetService,
+            recoveryListener,
+            repositoriesService,
+            onShardFailure,
+            globalCheckpointSyncer,
+            retentionLeaseSyncer);
+    }
+
     @Override
     public IndexShard createShard(
```

Comment:
Can we lose this override? It looks only to be used in one test that could use the default recovery state factory anyway.

Reply:
Do you mean removing this method from `AllocatedIndices` interface?

Reply:
Yes